nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -814,24 +812,33 @@ define([
         }
 
         /**
-         * Globally cancels all running jobs and the run status by the following steps.
-         * 1. If we've clicked run, but we don't have any jobs yet, just remove the
-         *    listener and ignore the jobs.
-         *    //TODO: make this wait for jobs to start first, then cancel them all in turn?
-         *      or set up some state that we catch jobs when they appear and THEN cancel them?
-         *      Maybe send a kernel message to stop all jobs with this run id?
-         * 2. If we have a list of jobs, cancel them all.
-         * 3. Return to the editing state, trigger updateEditingState.
+         * Either cancel or reset the app
+         *
+         * Cancel all running jobs and reset the cell for editing. Both actions
+         * have the same effect but the user dialogs shown differ.
+         *
+         * The user is shown a dialog to confirm their choice prior to cancellation
+         * and resetting.
+         * If a job run has been triggered, the batch job will be terminated, which
+         * will also terminate all the child jobs.
+         *
+         * @param {string} actionType - either 'cancel' or 'reset'; defaults to 'reset'
          */
-        async function doCancelCellAction() {
-            const confirmed = await DialogMessages.showDialog({ action: 'cancelBulkImport' });
+        async function doCellAction(actionType) {
+            if (!actionType) {
+                actionType = 'reset';
+            }
+            const action = actionType === 'cancel' ? 'cancelBulkImport' : 'appReset';
```

Comment:
Why not just allow `cancelBulkImport` or `appReset` and use the registered strings directly?

Reply:
Doh! Done, thanks for pointing out an obvious improvement!